--- a/views_bulk_operations.js	2010-11-13 00:34:33 +0000
+++ b/views_bulk_operations.js	2010-12-03 23:11:15 +0000
@@ -88,7 +88,7 @@
       url: Drupal.settings.basePath+'views-bulk-operations/js/pager',
       data: 'url='+escape(Drupal.settings.vbo.url)+'&selection='+selection+'&select_all='+select_all+'&page='+page[1],
       datatype: 'json',
-      async: false,
+      async: false
     });
     return true;
   });

