--- a/services.module	12 May 2010 19:17:06 -0000	1.8.2.88.2.18
+++ b/services.module	21 Jun 2010 18:07:39 -0000
@@ -832,9 +832,11 @@ function services_session_load($sessid) 
     return $user;
   }
 
-  // Make backup of current user and session data
+  // Make backup of current user and any existing session data
   $backup = $user;
-  $backup->session = session_encode();
+  if (!empty($_SESSION)) {
+    $backup->session = session_encode();
+  }
 
   // Empty current session data
   $_SESSION = array();
