diff --git a/automodal.module b/automodal.module
index d007ef8..32d9a5d 100755
--- a/automodal.module
+++ b/automodal.module
@@ -188,9 +188,17 @@ function automodal_perm() {
  * Implementation of hook_form_alter()
  */
 function automodal_form_alter(&$form, $form_state, $form_id) {
-  if (automodal_request_is_child() && filter_input(INPUT_GET, 'automodalClose', FILTER_VALIDATE_BOOLEAN)) {
+  if (!empty($form_state['storage']['automodal']) || (automodal_request_is_child() && filter_input(INPUT_GET, 'automodalClose', FILTER_VALIDATE_BOOLEAN))) {
     // Name the form submit entry so it can easily be found and replaced
     $form['#submit']['automodal'] = 'automodal_close_form_submit';
+
+    if (empty($form_state['storage']['automodal'])) {
+      $form_state['storage']['automodal'] = array(
+        'close'    => TRUE,
+        'reload'   => filter_input(INPUT_GET, 'automodalReload', FILTER_VALIDATE_BOOLEAN),
+        'redirect' => filter_input(INPUT_GET, 'automodalRedirect', FILTER_SANITIZE_STRING),
+      );
+    }
   }
 }
 
@@ -198,11 +206,14 @@ function automodal_form_alter(&$form, $form_state, $form_id) {
  * Implementation of hook_form_submit()
  */
 function automodal_close_form_submit($form, &$form_state) {
+  $automodal = $form_state['storage']['automodal'];
+  unset($form_state['storage']['automodal']);
+
   $args = array();
-  if (filter_input(INPUT_GET, 'automodalReload', FILTER_VALIDATE_BOOLEAN)) {
+  if ($automodal['reload']) {
     $args['reload'] = TRUE;
   }
-  if ($redirect = filter_input(INPUT_GET, 'automodalRedirect', FILTER_SANITIZE_STRING)) {
+  if ($redirect = $automodal['redirect']) {
     $args['redirect'] = $redirect;
   }
 
