diff --git a/drupal/sites/all/modules/quiz/question_types/long_answer/long_answer.classes.inc b/drupal/sites/all/modules/quiz/question_types/long_answer/long_answer.classes.inc
index 2ca1ee7..1001fe0 100644
--- a/drupal/sites/all/modules/quiz/question_types/long_answer/long_answer.classes.inc
+++ b/drupal/sites/all/modules/quiz/question_types/long_answer/long_answer.classes.inc
@@ -262,6 +262,11 @@ class LongAnswerResponse extends QuizQuestionResponse {
    */
   public function isValid() {
     if (trim($this->answer) == '') return t('You must provide an answer');
+
+    // DRS: limit words to 250!
+    $word_count = count(explode(' ', trim($this->answer)));
+    if ($word_count > 250) return t('Your answer must be less than 250 words (yours is %length)', array('%length' => $word_count));
+
     return TRUE;
   }
 
