=== modified file 'includes/admin.inc'
--- includes/admin.inc	2010-07-16 09:55:25 +0000
+++ includes/admin.inc	2010-09-16 01:30:34 +0000
@@ -1603,6 +1603,8 @@
       $stack = $view->stack; // copy so the next shift doesn't break the array
       $top = array_shift($stack);
       $top[0] = $js; // change identifier into $js setting
+      $stepview = $top[2]; // Change view into a reference [#452384]
+      $top[2] = &$stepview;
       $form_state = call_user_func_array('views_ui_build_form_state', $top);
       $form_state['input'] = array(); // this is a new form, make sure it
       // doesn't try to inherit $_POST info.

