=== modified file 'nodecomment.module'
--- nodecomment.module	2011-05-06 14:36:02 +0000
+++ nodecomment.module	2011-05-31 11:17:01 +0000
@@ -885,6 +885,11 @@
     $items['node/%node']['page callback'] = 'nodecomment_node_view';
   }
 
+  // Disable normal comment reply page
+  if (module_exists('comment')) {
+    $items['comment/reply/%node']['access callback'] = '_nodecomment_comment_reply_access';
+  }
+
   $comment_types = nodecomment_get_comment_types();
   foreach (node_get_types('names') as $type => $blank) {
     $path = "node/add/". str_replace('_', '-', $type);
@@ -951,6 +956,12 @@
   return node_access('create', $type);
 }
 
+function _nodecomment_comment_reply_access($op, $node) {
+  if ($node->comment_type) {
+    return FALSE;
+  }
+  return node_access($op, $node);
+}
 
 /**
  * Menu callback; view a single node.

